### PR TITLE
DDF-4475 Add additional extensibility to result item

### PR DESCRIPTION
**Pull request by andrewzimmer**
#### What does this PR do?
Add additional extensibility to item-result buttons for use in downstream projects
#### Who is reviewing it? 
@hayleynorton @Bdthomson 
#### Select relevant component teams: 
@codice/ui 
#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining
@djblue
#### How should this be tested?
- build/run
- ingest data
- verify no regression for result items (mostly buttons)
#### What are the relevant tickets?
[](https://codice.atlassian.net/browse/)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/component/result-item/result-item.view.js [hunk 1]
```diff
@@ -185,6 +185,9 @@ const ResultItemView = Marionette.LayoutView.extend({
                 title="Follow external links"
                 data-help="Follow external links."
               />
+              <div className="result-validation">
```
Comment (andrewkfiedler) [inline ui/packages/catalog-ui-search/src/main/webapp/component/result-item/result-item.view.js]: Why do we need the className here?

Comment (andrewzimmer) [reply ui/packages/catalog-ui-search/src/main/webapp/component/result-item/result-item.view.js]: My extension only returns a button component, so there is some styling needed here for it to look correct. Should this div and styling be added in the extension instead?

**Comment by Bdthomson** [comment]
build now

**Comment by Bdthomson** [comment]
build now (jenkins allocation failure)

**Comment by Bdthomson** [comment]
Hero in Progress

**Pull request merged**